integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/PlanUtils3.java

```diff
@@ -78,10 +98,26 @@ public static <D extends OpenLineage.Dataset> List<D> fromDataSourceV2Relation(
       OpenLineageContext context,
       DataSourceV2Relation relation,
       OpenLineage.DatasetFacetsBuilder datasetFacetsBuilder) {
+
+    OpenLineage openLineage = context.getOpenLineage();
+
+    Optional<DatasetIdentifier> di;
     // Get identifier for dataset, or return empty list
     if (relation.identifier().isEmpty()) {
       log.warn("Couldn't find identifier for dataset in plan {}", relation);
-      return Collections.emptyList();
+      di = PlanUtils3.getDatasetIdentifier(context, relation);
+      if (!di.isPresent()) {
+        return Collections.emptyList();
+      } else {
+
+        datasetFacetsBuilder
+            .schema(PlanUtils.schemaFacet(openLineage, relation.schema()))
+            .dataSource(PlanUtils.datasourceFacet(openLineage, di.get().getNamespace()));
+
+        return Collections.singletonList(
```

Comment:
Do we need to return datasets like this? Wouldn't it be possible to include `else` implementation for `if (relation.identifier().isEmpty())` condition to establish `DatasetIdentifier` in a different way, and then return dataset facets list.

Reply:
I moved the catalog handler specific code inside the `else` condition for `(relation.identifier().isEmpty())`, and so we have 

    ` datasetFacetsBuilder
        .schema(PlanUtils.schemaFacet(openLineage, relation.schema()))
        .dataSource(PlanUtils.datasourceFacet(openLineage, di.get().getNamespace()));
    
    return Collections.singletonList(
        datasetFactory.getDataset(
            di.get().getName(), di.get().getNamespace(), datasetFacetsBuilder.build())); `

Common to both the RelationHandler and CatalogHandler cases.

Reply:
Seems like some debug logging got committed. Please remove it.

Reply:
This one `log.warn("Couldn't find identifier for dataset in plan {}", relation);`? In the previous code this was logged when no identifier was found. Should we remove it or switch it to `info` level?

Reply:
This one is fine. I think I saw some debugging logs. 
Anyway, they're not there anymore.